app/models/comment.rb

```diff
@@ -1,4 +1,14 @@
 class Comment < ApplicationRecord
-  belongs_to :user
-  belongs_to :post
+  belongs_to :user, class_name: 'User'
+  belongs_to :post, class_name: 'Post'
+
+  after_create :update_comment_counter_post
+
+  def author=(user)
+    self.user_id = user.id
+  end
+
+  def update_comment_counter_post
+    post.update(comments_counter: post.comments.count)
+  end
```

Comment:
## Optional
- [ ] I suggest that you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `#update_comment_counter_post` method every time a new comment has been saved to the database, then make the `#update_comment_counter_post` method private